tests/nvmeof/test_ceph_nvmeof_qos_tests.py

```diff
@@ -0,0 +1,344 @@
+                        **{
+                            "args": qos_arg,
+                        }
+                    )
+                    verify_qos(qos_arg, nvmegwcli)
+                    prev_key = key
+
+                    run_io_and_validate_qos(
+                        ceph_cluster, config, key, qos_args, nvmegwcli
+                    )
+
+            # Update QoS values
+            qos_arg = {
+                "nsid": qos_args["nsid"],
+                "subsystem": qos_args["subsystem"],
+                "r-megabytes-per-second": 10,
+                "w-megabytes-per-second": 20,
+                "rw-megabytes-per-second": 30,
+                "rw-ios-per-second": 2000,
+            }
+            nvmegwcli.namespace.set_qos(
+                **{
+                    "args": qos_arg,
+                }
+            )
+            verify_qos(qos_arg, nvmegwcli)
+
+
+def disconnect_initiator(ceph_cluster, node, subnqn):
+    """Disconnect Initiator."""
+    node = get_node_by_id(ceph_cluster, node)
+    initiator = Initiator(node)
+    initiator.disconnect(**{"nqn": subnqn})
+
+
+def disconnect_all_initiator(ceph_cluster, nodes):
+    """Disconnect all connections on Initiator."""
+    for node in nodes:
+        node = get_node_by_id(ceph_cluster, node)
+        initiator = Initiator(node)
+        initiator.disconnect_all()
+
+
+def teardown(ceph_cluster, rbd_obj, nvmegwcli, config):
+    """Cleanup the ceph-nvme gw entities.
+
+    Args:
+        ceph_cluster: Ceph Cluster
+        rbd_obj: RBD object
+        config: test config
+    """
+    # Delete the multiple Initiators across multiple gateways
+    if "initiators" in config["cleanup"]:
+        for initiator_cfg in config["initiators"]:
+            disconnect_initiator(
+                ceph_cluster, initiator_cfg["node"], initiator_cfg["nqn"]
+            )
+
+    if "disconnect_all" in config["cleanup"]:
+        nodes = config.get("disconnect_all")
+        if not nodes:
+            nodes = [i["node"] for i in config["initiators"]]
+
+        disconnect_all_initiator(ceph_cluster, nodes)
+
+    # Delete the multiple subsystems across multiple gateways
+    if "subsystems" in config["cleanup"]:
+        config_sub_node = config["subsystems"]
+        if not isinstance(config_sub_node, list):
+            config_sub_node = [config_sub_node]
+        for sub_cfg in config_sub_node:
+            node = config["gw_node"] if "node" not in sub_cfg else sub_cfg["node"]
+            LOG.info(f"Deleting subsystem {sub_cfg['nqn']} on gateway {node}")
+            nvmegwcli.subsystem.delete(
+                **{"args": {"subsystem": sub_cfg["nqn"], "force": True}}
+            )
+
+    # Delete the gateway
+    if "gateway" in config["cleanup"]:
+        delete_nvme_service(ceph_cluster, config)
+
+    # Delete the pool
+    if "pool" in config["cleanup"]:
+        rbd_obj.clean_up(pools=[config["rbd_pool"]])
+
+
+def trigger_fio(ceph_cluster, config, io_mode, key):
```

Comment:
suggesting to move it under utils module.

Reply:
This was very specific to every QoS setting(like L208), I don't think it makes sense if I have this in utils.